bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/communication/RemoteAPIController.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.airgradient.internal.communication;
+
+import static org.openhab.binding.airgradient.internal.AirGradientBindingConstants.CONTENTTYPE_JSON;
+import static org.openhab.binding.airgradient.internal.AirGradientBindingConstants.CONTENTTYPE_OPENMETRICS;
+import static org.openhab.binding.airgradient.internal.AirGradientBindingConstants.CONTENTTYPE_TEXT;
+import static org.openhab.binding.airgradient.internal.AirGradientBindingConstants.REQUEST_TIMEOUT;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.airgradient.internal.config.AirGradientAPIConfiguration;
+import org.openhab.binding.airgradient.internal.handler.AirGradientAPIHandler;
+import org.openhab.binding.airgradient.internal.model.LedMode;
+import org.openhab.binding.airgradient.internal.model.Measure;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Helper for doing rest calls to the AirGradient API.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ */
+@NonNullByDefault
+public class RemoteAPIController {
+
+    private final Logger logger = LoggerFactory.getLogger(RemoteAPIController.class);
+
+    private final HttpClient httpClient;
+    private final Gson gson;
+    private final AirGradientAPIHandler handler;
+    private final AirGradientAPIConfiguration apiConfig;
+
+    public RemoteAPIController(HttpClient httpClient, Gson gson, AirGradientAPIHandler handler,
```

Comment:
Great to see that you improve the binding and take the effort to seperate these classes. Now we have one more thing todo. The `RemoteAPIController` should not be aware of the `AirGradientAPIHandler` class.
That can be achieved by removing this parameter in the constructor and communicate about status updates by throwing an exception or not.

Reply:
Thanks! Yes, that was much nicer